src/module.c

```diff
@@ -3194,6 +3194,22 @@ int VM_ReplyWithError(ValkeyModuleCtx *ctx, const char *err) {
     return VALKEYMODULE_OK;
 }
 
+static int moduleReplyErrorFormatInternal(ValkeyModuleCtx *ctx, int flags, const char *fmt, va_list ap) {
+    client *c = moduleGetReplyClient(ctx);
+    if (c == NULL) return VALKEYMODULE_OK;
+
+    int len = strlen(fmt) + 2; /* 1 for the \0 and 1 for the hyphen */
+    char *hyphenfmt = zmalloc(len);
```

Comment:
Is it better to put it on the stack?

```suggestion
    char hyphenfmt[len];
```

Reply:
Can you do that? How the compiler knows the size of `hyphenfmt` at compile time?

Reply:
I changed the code with your suggestion and it works :exploding_head:

Reply:
I need to check the assembly code to understand what's going on :smile:

Reply:
Ah, it's a C99 feature. Variable Length Arrays. The things we learn after all these years :laughing:

Reply:
It's like the old time `alloca()`. Btw, C99 came out 26 years ago. :)